### PR TITLE
Add latency metrics to python device app

### PR DESCRIPTION
**Pull request by BertKleewein**

### CODE REVIEW

File: docs/metrics.md [hunk 1]
```diff
@@ -11,6 +11,16 @@ Not all metrics are sent to all destinations.
...
 | System Health Metrics | - | X | X |
+| Latency Metrics | - | - | X |
+
+## Metric name casing
+
+When metrics are set in json (for reported properties and telemetry), the name is written using camelCase.
+When metrics are pushed to Azure monitor, the name is written using PascalCase.
+
+For example, the metric `receiveC2dCountReceived` is named:
+* `receiveC2dCountReceived` with a lower case `r` when used in desired properties and telemetry, an
```
Comment (drwill-ms) [inline docs/metrics.md:22]: an = and?

File: docs/metrics.md [hunk 2]
```diff
@@ -112,6 +122,17 @@ Most or all of the test metrics overlap with metrics that are sent in telemetry
...
 
+## Latency metrics
+Latency metrics are only pushed to Azure Monotor.
+
+| metric name | format | meaning |
+| - | - | - |
+| `LatencyQueueMessageToSendInMilliseconds` | float | Number of milliseconds between a message is queued to send and when it is actually sent. |
+| `LatencySendMessageToServiceAckInSeconds` | float | Number of seconds between when a message is sent and when the corresponding `serviceAck` is received back from the service. |
+| `LatencyAddReportedPropertyToServiceAckInSeconds` | float | Number of seconds between when a reported property is added and when the corresponding `serviceAck` is received back from the service. |
+| `LatencyRemoveReportedPropertyToServiceAckInSeconds` | float | Number of seconds etween when a reported property is removed and when the corresponding `serviceAck` is received back from the servie. |
```
Comment (drwill-ms) [inline docs/metrics.md:133]: misspelled:
- between
- service

File: docs/metrics.md [hunk 3]
```diff
@@ -112,6 +122,17 @@ Most or all of the test metrics overlap with metrics that are sent in telemetry
...
+| `LatencyBetweenC2dInSeconds` | float | Number of seconds between consecuitive C2d messages. |
```
Comment (drwill-ms) [inline docs/metrics.md:134]: misspelled:
consecutive

File: python/device/device.py [hunk 4]
```diff
@@ -269,6 +297,32 @@ def _configure_azure_monitor_metrics(self):
             "patches with remove operations(s)",
         )
 
+        self.reporter.add_float_measurement(
+            MetricNames.LATENCY_QUEUE_MESSAGE_TO_SEND,
+            "Number of milliseconds between queueing a message and actually sending it",
```
Comment (drwill-ms) [inline python/device/device.py:302]: misspelled queuing

File: docs/metrics.md (hunk 3 above)
Comment (BertKleewein) [reply docs/metrics.md:134]: I need to get an editor with a built-in spell checker. :)

File: python/device/device.py [hunk 5]
```diff
@@ -269,6 +297,32 @@ def _configure_azure_monitor_metrics(self):
...
+            "seconds",
+        )
+        self.reporter.add_float_measurement(
+            MetricNames.LATENCY_REMOVE_REPORTED_PROPERTY_TO_SERVICE_ACK,
+            "Number of seconds between clearing a reported property and receiving the serviceAck back from the service app",
+            "seconds",
+        )
+        self.reporter.add_float_measurement(
+            MetricNames.LATENCY_BETWEEN_C2D,
+            "number of seconds between test c2d messages from the service",
```
Comment (drwill-ms) [inline python/device/device.py:322]: capital N in number

**Review by drwill-ms — APPROVED** [review]

File: python/device/device.py (hunk 4 above)
Comment (BertKleewein) [reply python/device/device.py:302]: Interesting.  For some twisted reason, this is the comment I chose to dig into the most.

* Oxford English Dictionary says "queueing", 
* Merriam-Webster says "queuing or queueing"
* Dictionary.com says "queuing"
* The Cambridge Dictionary says "queueing"
* Wikipedia says (at https://en.wikipedia.org/wiki/Queueing_theory):
> The spelling "queueing" over "queuing" is typically encountered in the academic research field. In fact, one of the flagship journals of the profession is Queueing Systems.

I think the snobs are on my side on this one. :)

**Pull request merged**